htdocs/core/class/extrafields.class.php

```diff
@@ -1860,8 +1860,7 @@ function setOptionalsFromPost($extralabels, &$object, $onlykey='')
 				if ($this->attributes[$object->table_element]['required'][$key])	// Value is required
 				{
 					// Check if empty without using GETPOST, value can be alpha, int, array, etc...
-					if ((! is_array($_POST["options_".$key]) && empty($_POST["options_".$key]) && $_POST["options_".$key] != '0')
-						|| (is_array($_POST["options_".$key]) && empty($_POST["options_".$key])))
+					if (empty($_POST["options_".$key]))
```

Comment:
If value is '0', empty will be true, so it will return an error.
For a combo list with multiple choice with first choice that has ke '0' and label "My value 0" will not be possible to be selected. Old code seems good.

Reply:
I reverted to the old check but I don't really understand your use case. The tooltips for `checkbox`- `select`-typed extrafields say that the keys should not be '0', so that Dolibarr itself could propose the empty choice with '0' key and empty label. Is the tooltip incorrect ?

Still, making the '0' choice available if the field is required seems an issue to me.

Reply:
The "tooltips" are better developers. They suggest to use 0 to avoid problems we comments. So it is better.
But i am not sure that everybody will follow advice and if a can keep the maximum of compatibility/working features, even if advices is not followed, it is better. But may be one day, we will just forbidden use of 0 as a key. In such a case, there is no more problem, but for the moment it is just a tooltip, not enough to rely on it.